packages/swing-store-lmdb/src/lmdbSwingStore.js

```diff
@@ -164,67 +165,213 @@ function makeSwingStore(dirPath, forceReset = false) {
     delete: del,
   };
 
+  /** @type {Set<number>} */
+  const activeStreamFds = new Set();
+  /** @type {Map<string, number>} */
+  const streamFds = new Map();
+  /** @type {Map<string, string>} */
+  const streamStatus = new Map();
+  let statusCounter = 0;
+
+  const STREAM_START = harden({ offset: 0, itemCount: 0 });
+
+  function insistStreamName(streamName) {
+    assert.typeof(streamName, 'string');
+    assert(
+      streamName.match(/^[-\w]+$/),
+      X`invalid stream name ${q(streamName)}`,
+    );
+  }
+
+  function closefd(fd) {
+    try {
+      fs.closeSync(fd);
+    } catch (e) {
+      // closing an already closed fd is OK, but any other errors are probably bad
+      if (e.code !== 'EBADF') {
+        throw e;
+      }
+    }
+  }
+
+  /**
+   * Close a stream that's open for read or write.
+   *
+   * @param {string} streamName  The stream to close
+   */
+  function closeStream(streamName) {
+    insistStreamName(streamName);
+    const fd = streamFds.get(streamName);
+    if (fd) {
+      closefd(fd);
+      streamFds.delete(streamName);
+      activeStreamFds.delete(fd);
+      streamStatus.set(streamName, 'unused');
```

Comment:
Hm? This consumes (admittedly a small amount of) RAM for every stream that's ever been referenced in the process' lifetime, right? Do we need to remember it after it's been closed?

Reply:
I was approaching this from the "do the straightforward but not necessarily efficient thing and then revisit once it's working" perspective, but since we have to deal with the uninitialized stream case anyway, it may make more sense to encode an unused stream by deleting the status entry.  I will do that.